### ISSUE TITLE
Improve "out of the box" experienced for FMU generation in OMEdit

### ISSUE THREAD

**Issue by dietmarw** [OpenModelica/OpenModelica#10174]
### Description

On a fresh system with OMEdit installed the export of cross-compiled FMUs will fail even if docker is already installed. This is  because the default settings for the cross-compiled FMU targets specify `--pull=never`

### Steps to Reproduce

1. Have a fresh system with only docker and OMEdit installed
2. activate one of the cross-compilation targets in "Tools -> Options -> FMI"
3. select a model to export as FMU
4. Export will fail with `docker: Error response from daemon: No such image: multiarch/crossbuild:latest.`

### Expected Behavior

If `multiarch/crossbuild` has not been pulled before then OMEdit should simply let docker pull that missing image so it can operate. Now I guess there was a reason why the `--pull=never` option is built in but I really don't see the point. That image is not updated that frequently and if yes, then there is a good reason for it which also means it should be updated in order to generate the FMU.
Alternatively the user should *at least* get the information of what to do in order to get hold of the image. Many users are not familiar with the docker technology but might be able to at least install docker itself on the system. 

To summarise, OMEdit should either 
1. (preferred) remove the `--pull=never` options and let docker deal with keeping the image up to date or not.
2. display the information what to do in order to get hold of the needed `multiarch/crossbuild` image.

### Version and OS

 - OpenModelica Version:  1.21.0~dev-233-geac10d9
 - OS: Windows 10 and Linux

**Comment by AnHeuermann** [comment]
> If multiarch/crossbuild has not been pulled before then OMEdit should simply let docker pull that missing image so it can operate. Now I guess there was a reason why the --pull=never option is built in but I really don't see the point. 

This was a design decision. It is very easy to build malicious docker containers and publish them on the forums or any other source. So if I convince you to run a mos script like

```modelica
loadModel(Modelica);
buildModelFMU(Modelica.Blocks.Examples.PID_Controller, platforms={""x86_64-linux-gnu docker run mulltiarch/crossbuild""}); // <-- notice the "typo" in the docker image pointing to a non-existing repo
```

I may could get root access to your machine, see also https://stackoverflow.com/questions/41991905/docker-root-access-to-host-system.
By forcing you to manually install new docker images you have a second chance to notice that you should already have this image and that something is wrong.
Of course one can argue if this is really more secure since Modelica models can already run arbitrary code, but usually not as root.

But a message how to install new images as well as a warning about risks is a good idea.

**Comment by dietmarw** [comment]
I see. But the danger of installing malicious docker images is when the user is doing it based on wrong instructions, right? So if OpenModelica does it with the curated correct image destination, then this is not a problem plus the user has something that actually works. Simply dropping the `--pull=never` would do exactly that.

**Comment by bilderbuchi** [comment]
I appreciate this safety-by-design approach with `--pull=never`. At the same time, some hint about what to do would be useful.

I think it also useful that this "teaches" a safe approach and awareness about this. Simply dropping  `--pull=never`, while safe for the curated built-in images, would not do that, and poses the danger of copy-paste-and-mutating the built in strings and normalising this pattern.

**Comment by bilderbuchi** [comment]
Another "fun" aspect concerning the out-of-the-box experience is that the `crossbuild` image is based on such an old Debian that it ships an ancient CMake, with the consequence of clashing with current default flags
> CMake Error at CMakeLists.txt:31 (message):
  --fmuRuntimeDepends=modelica requires CMake version 3.21 or higher.   You are running version 3.7.2.   Use OpenModelica compiler flag '--fmuRuntimeDepends=none' to disable
